Services/CompanyService.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using DatabaseAccess.Models;
+using DatabaseAccess.UOW;
+
+
+namespace Services
+{
+    public class CompanyService
+    {
+
+        public IList<Company> GetAllCompanies()
+        {
+            using (UnitOfWork uow = new UnitOfWork())
+            {
+                return uow.CompanyRepository.GetAll();
+            }
+        }
+
+        public void UpdateCompany(Company company)
+        {
```

Comment:
o sa apara aceeasi problema pe care a avut-o si toma (nu stiu daca stii de ea :D dar l-am rugat sa te ajute), pentru ca iei in if obiectul din db, el e "urmarit" de context, iar apoi cand incerci sa ii faci update o sa dea o eroare, cum ca obiectul e "urmarit" deja. si atunci cel mai bine e sa iei obiectul din db intr-o variabila, sa verifici daca e null sau nu, apoi sa schimbi fieldurile obiectului din db la fieldurile obiectului primit, apoi faci update pe obiectul din db. am pus poza mai jos cu varianta finala de la toma

![image](https://user-images.githubusercontent.com/36417662/47968349-67129f00-e071-11e8-9145-dd416755c955.png)